netmiko/cisco/cisco_xr.py

```diff
@@ -105,36 +114,74 @@ def commit(
         else:
             command_string = "commit"
 
+        if force:
+            command_string = command_string.replace("commit", "commit force")
+        if best_effort:
+            command_string = command_string.replace("commit", "commit best-effort")
+        if replace:
+            command_string = command_string.replace("commit", "commit replace")
+        print(f"command string is {command_string}")
+
         # Enter config mode (if necessary)
         output = self.config_mode()
 
-        # IOS-XR might do this:
-        # This could be a few minutes if your config is large. Confirm? [y/n][confirm]
-        new_data = self._send_command_str(
-            command_string,
-            expect_string=r"(#|onfirm)",
-            strip_prompt=False,
-            strip_command=False,
-            read_timeout=read_timeout,
-        )
-        if "onfirm" in new_data:
-            output += new_data
+        if replace:
+            new_data = self._send_command_str(
+                command_string,
+                expect_string=r"This commit will replace or remove the entire running configuration",
```

Comment:
i think we should change all the expected pattern everywhere and provide complete patterns instead of either just the start of pattern or end. this is to make sure router actually prints whole pattern and not get stuck. I am trying to be more stricter on XR based on our VXR experience

in this case, it will be below.
RP/0/RP0/CPU0:ios(config)#commit repl
Tue Apr  4 04:03:52.215 UTC

This commit will replace or remove the entire running configuration. This
operation can be service affecting.
Do you wish to proceed? [no]:

Reply:
I understand why you are suggesting this.
This pattern checking is more aligned with the origin netmiko, and would it require to make sure that the complete big string is same for all the pids, XR flavors  and architecture?